public/components/QueryEditor/QueryEditor.tsx

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useMemo, useRef } from 'react';
+import { monaco } from '@osd/monaco';
+import { useDispatch, useSelector } from 'react-redux';
+import { DEFAULT_DATA } from '../../../../../src/plugins/data/common';
+import { selectQueryLanguage, selectIsQueryEditorDirty } from '../../redux/selectors';
+import { setIsQueryEditorDirty, setDataset } from '../../redux/slices';
+
+type Editor = monaco.editor.IStandaloneCodeEditor;
+
+// Match explore plugin's trigger characters exactly
```

Comment:
Not blocking:
The alerting plugin has it's own meaning for the term "trigger." In a future PR, could you expand on this comment to explain what explore is using these trigger characters for, and how we're using them?